processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -38,7 +38,11 @@ public int compare(String str1, String str2)
     {
       int[] pos = {0, 0};
 
-      if (str1.length() == 0) {
+      if (str1 == null) {
+        return str2 == null ? 0 : -1;
```

Comment:
@himanshug I can do that but don't you think that its better to change LexicographicTopNMetricSpec compare implementation to return 0 when both values are null. Anyways, it should not matter as the condition for adding entry is >0  as of now.